Sources/PriorityQueue/PriorityQueue.swift

```diff
@@ -0,0 +1,227 @@
+     *  @param  index  a valid index of a certain element
+     *  @return  the index of a certain element's right child, or -1 if the element has no right child
+     */
+    private func getRightChild(index:Int) -> Int{
+        let child = (2*index)+2
+        return (child >= size()) ? -1 : child;
+    }
+    
+    /*
+     *  Removes the item with highest/lowest priority in the Priority Queue
+     *  @return  the item with highest/lowest priority or nil if the Priority Queue is empty
+     */
+    public mutating func dequeue() -> Element?{
+        if(isEmpty() == true){
+            return nil;
+        }
+        let removedElement = storage[0];
+        storage[0] = storage.last!
+        storage.removeLast()
+        sinkDown(elementIndex: 0)
+        return removedElement;
+    }
+    
+    /*
+     *  Moves a specified element up the array (away from index zero) until it's in its natural position
+     */
+    private mutating func sinkDown(elementIndex:Int){
+        var currentIndex = elementIndex
+        var leftChild = getLeftChild(index: elementIndex)
+        var rightChild = getRightChild(index: elementIndex)
+        var childToSwap:Int
+        var temp:Element
+        
+        //loops until a swap is not needed or currentIndex has no children
+        while(true){
+            //chooses which child to swap currentIndex with
+            if(leftChild != -1 && rightChild != -1){
+                //both children are present
+                childToSwap = (compare(firstElement: storage[leftChild], secondElement: storage[rightChild]) < 0) ? leftChild : rightChild;
+            }else if(leftChild != -1){
+                //only left child present
+                childToSwap = leftChild
+            }else if(rightChild != -1){
+                //only right child present
+                childToSwap = rightChild
+            }else{
+                //neither children present
+                break;
+            }
+            //swap currentIndex with appropriate child if needed
+            if(compare(firstElement: storage[childToSwap], secondElement: storage[currentIndex]) < 0){
+                temp = storage[currentIndex]
+                storage[currentIndex] = storage[childToSwap]
+                storage[childToSwap] = temp
+                
+                //updates children and current index in case swap caused more problems
+                currentIndex = childToSwap;
+                leftChild = getLeftChild(index: childToSwap)
+                rightChild = getRightChild(index: childToSwap)
+            }else{
+                //swap is not needed
+                break;
+            }
+        }
+        
+        
+    }
+    
+    /*
+     *  Determines the element with the highest/lowest priority, depending on the heap type
+     *  @return  the element with the highest/lowest priority, or nil if no elements exist
+     */
+    public func peek() -> Element?{
```

Comment:
I think it would be better to reuse the `min()`/`max()` member names from the `Sequence` protocol.

Reply:
I noted to change this in the comments, once Heap.swift allows for retrieving both the min **and** max values, instead of just one of them.